source/binarySearch.cpp

```diff
@@ -79,56 +78,110 @@ uint32_t BinarySearchTree::qualSum(char *q) {
+
+
+
+bool BinarySearchTree::FlipBitsCheck(char *seq, bool r2) {
+    /*Another error check for the humans*/
+    if (strlen(seq) < start+charLength) {
+        fprintf(stderr, "Error within binarySearch.cpp in funciton FlipBitsChars() strlen of %s larger than start + length\n", r2 ? "R2" : "R1");
+        fprintf(stderr, "String Len = %lu, start = %d, length = %d\n", strlen(seq), start, charLength);
+        disReads++;
+        return false;
+    }
+
+    return true;
+
+}
+
+
+char RC_BP(char bp) {
+    if (bp == 'N') {
+        return 'N';
+    } else if (bp == 'A') {
+        return 'T';
+    } else if (bp == 'T') {
+        return 'A';
+    } else if (bp == 'G') {
+        return 'C';
+    } else if (bp == 'C') {
+        return 'G';
+    } else {
+        return '\0';
+    }
+}
+
+void RC_Read(char **seq) {
+    char *new_seq = strdup(*seq);
+    int len = strlen(*seq);
+    int i = 0;
+    while ((*seq)[i] != '\0') {
+        (*seq)[(len - 1) - i] = RC_BP(new_seq[i]);
+        if ((*seq)[(len -1) - i] == '\0') {
+            fprintf(stderr, "ERROR in RC_Read in binarySearch.cpp, Illegal char in %s\n", *seq);
+            exit(42);
+        }
+        i++;
+    }
+    free(new_seq);
+}
+
+
+
 /*Flip bits functionality for character string*/
-bool BinarySearchTree::FlipBitsChars(readInfo *R1, readInfo *R2, uint16_t **id) {
+bool BinarySearchTree::FlipBitsChars(readInfo *R1, readInfo *R2, uint16_t **id, bool RC) {
 
-    uint16_t loc = start;
     char *seq_1, *seq_2;
     
     /*Incementor for id array*/
     uint16_t idLoc = 0, bitShifts = 0;
 
     seq_1 = R1->getSeq();
+    if (!(FlipBitsCheck(seq_1, false))) { return false; }
 
     if (R2) {
         seq_2 = R2->getSeq();
+        if (!(FlipBitsCheck(seq_2, true))) { return false; }
     } else {
         seq_2 = NULL;
     }
+    
+    char *seq;
 
+    if (seq_2 == NULL) {
+        seq = (char *)malloc(sizeof(char) * charLength + 1);
+        sprintf(seq, "%.*s", charLength, seq_1 + start);
+    } else {
+        seq = (char *)malloc(sizeof(char) * charLength * 2 + 1 );
+        sprintf(seq, "%.*s%.*s", charLength, seq_1 + start, charLength, seq_2 + start);
+    }
 
-    /*Another error check for the humans*/
-    if (strlen(seq_1) < start+charLength) {
-        fprintf(stderr, "Error within binarySearch.cpp in funciton FlipBitsChars() strlen of R1 larger than start + length\n");
-        fprintf(stderr, "String Len = %lu, start = %d, length = %d\n", strlen(seq_1), start, charLength);
-        disReads++;
-        return false;
+
+    if (RC) {
+        RC_Read(&seq);
     }
 
-    char *seq = seq_1;
-    
-    while (loc < start+charLength) {
-        uint16_t tmp;
-        if (seq[loc] == 'A') {
-            tmp = A;
-        } else if (seq[loc] == 'T') {
-            tmp = T;
-        } else if (seq[loc] == 'G') {
-            tmp = G;
-        } else if (seq[loc] == 'C') {
-            tmp = C;
-        } else if (seq[loc] == 'N') {
+
+    uint16_t loc = 0;
+
+    (*id)[0] = 0;
```

Comment:
Yeah, newb mistake, I think the results should be the same now.

I zero the array in the loop below, however, never the zeroth index.

Mind checking this out now @samhunter ?